### PR TITLE
Move testing outside training loop

### LINKED ISSUE #108: Is testing necessary in every epoch (MAML vision examples)?
I might be misunderstanding something completely but, in the MAML vision examples (maml_miniimagenet & maml_omniglot) the normal train/val/test split takes place but testing also happens in every epoch during training. Why is that? Isn't this the purpose of the validation set? Shouldn't testing be left just for after the training? Also, testing at every epoch slows down the training time.

### PR DESCRIPTION
**Pull request by Kostis-S-Z**
### Description

Fixes #108 

In Vision (omniglot & miniimagenet) examples in MAML, testing was taking place in every training iteration (which is redundant & adds to computational cost).

Fix: Move it outside.

### CODE REVIEW

**Comment by seba-1511** [comment]
Thanks for the PR @Kostis-S-Z .

The test that failed timed-out when downloading tiered-ImageNet. I've relaunched it, and as soon as it passes we can merge this in.
